src/components/DynamicFormContainer/DynamicFormLayout.tsx

```diff
@@ -44,11 +45,12 @@ export const DynamicFormLayout: FunctionComponent = () => {
   };
 
   const removeCurrentForm = (): void => {
-    // Remove current form data before going back
+    if (activeFormIndex === undefined) return;
     const nextForms = [...forms];
-    nextForms.splice(nextForms.length - 1, 1);
+    nextForms.splice(activeFormIndex, 1);
```

Comment:
I don't an answer for this, but just imagining if this process makes more sense(can ask around):

1. I add doc-1
2. I add doc-2
3. I add doc-3
4. I go back to doct-2 by clicking on the previous document arrow
5. I click on <- Back and get prompted with `Do you want to go back? This will delete ALL your current document.`
6. I click on "Confirm" and all documents gets deleted (Otherwise if I want to add more document, I click on cancel, then add new document)

Reply:
ok